File: stack.java
package test;

import java.util.Arrays;
import java.util.Scanner;

public class test {
    static Scanner sc = new Scanner(System.in);
    public static int peek(int[] arr, int size){
        if(isEmpty(arr)){
            return 0;
        }
        int index = index(arr, 0);
        return index;
    }
    public static boolean isFull(int size, int[] arr){
        int sizes = 0;
        for(int i = 0; i < size; i++){
            if(arr[i] != 0){
                sizes++;
            }
        }
        if(sizes == size){
            System.out.println("is full");
            return true;
        }
        System.out.println("is not full");
        return false;
    }

    public static boolean isEmpty(int[] arr){
        for(int i = 0; i < arr.length; i++){
            if(arr[i] !=0){
                System.out.println("is not empty");
                return false;
            }
        }
        System.out.println("is empty");
            return true;
    }

    public static int index(int[]arr, int size){
        int index = 0;
        for(int i = 0; i < size; i++){
            if(arr[i] != 0){
                index++;
            }
        }
        return index;
    }

    public static int[] pop(int[]arr, int size){
        if(isEmpty(arr)){
            return arr;
        }
        arr[index(arr, size) -1] = 0;
        return arr;
    }


    public static int[] push(int[]arr, int size){
            if(isFull(size, arr)){
            return arr;
        }
            try {
                System.out.println("enter a number: ");
                int numb = sc.nextInt();
                arr[index(arr, size)] = numb;
            } catch (Exception e) {
                System.out.println("invalid");
            }
        return arr;
    }

    public static void main(String[]args){
        int size = 5;
        int[] arr = new int[size];
        boolean run = true;
        while(run){
            System.out.println(Arrays.toString(arr));
                while(true){
                    System.out.print("""
                    #######################
                    1|peek
                    2|isEmpty
                    3|isFull
                    4|pop
                    5|push
                    #######################
                    """);
                    System.out.print("enter choice: ");
                    char ch = sc.next().charAt(0);
                    if(Character.isDigit(ch)){
                        int numb = Character.getNumericValue(ch);
                    switch(numb){
                        case 1: 
                        System.out.println("#######################");
                        System.out.println("top is "+peek(arr,size));
                            break;
                        case 2: 
                            System.out.println("#######################");
                            isEmpty(arr);
                            break;
                        case 3: 
                            System.out.println("#######################");
                            isFull(size, arr);
                            
                            break;
                        case 4: System.out.println("#######################");
                            arr=pop(arr, size);
                            break;
                        case 5: System.out.println("#######################");
                            arr = push(arr, size);
                            break;
                        default:
                            System.out.println("invalid");
                            continue;
                    }
                    break;
                    }else{
                        System.out.println("invalid");
                        sc.nextLine();
                    }
            }
            while(true){
                System.out.print(" Exit Y|N:");
                char exit = Character.toLowerCase(sc.next().charAt(0));

                if (Character.isDigit(exit)) {
                    System.out.println("Invalid input. Numbers not allowed.");
                    continue;
                }
                if (exit == 'y') {
                    System.out.println("Exiting...");
                    run = false;
                    break;
                } 
                if (exit == 'n') {
                    System.out.println("Continuing...");
                    break;
                }
                System.out.println("Invalid input. Please type Y or N.");
            }
    }
        sc.close();
    }
}

File: stack.py
def isEmpty(stack:list):
    if len(stack) == 0: 
        print("stack is empty")
        return True
    print("stack is not empty")
    return False

def isFull(stack:list, size):
    if len(stack) >= size:
        print("stack is full")
        return True
    print("stack is not full")
    return False

def pop(stack:list, top):
    if isEmpty(stack): return top
    stack.pop()
    top-=1
    return top

def push(stack:list, size, top):
    if isFull(stack, size): return top
    while True:
        ch = input("enter a value: ")
        if(ch.isdigit()): break;
        else: print("invalid input")
    
    ch = int(ch)
    top += 1
    stack.append(ch)
    return top
    
def peek(stack:list,top):
    if not isEmpty(stack): return stack[top]

def sizeA(stack:list):
    if not isEmpty(stack): return len(stack)
    
    
    
top = -1
size = 5;
stack = [];
while True:
    print(stack)
    print("\n***************")
    print("1 | isEmpty")
    print("2 | isFull")
    print("3 | pop")
    print("4 | push")
    print("5 | peek")
    print("6 | size")
    print("0 | exit")
    print("***************")
    ch = input("enter a choice: ")
    if not ch.isdigit(): print("invalid input");continue
    ch = int(ch)
    match(ch):
        case 0: print("exiting"); break
        case 1: isEmpty(stack)
        case 2: isFull(stack, size)
        case 3: top = pop(stack, top)
        case 4: top = push(stack, size, top)
        case 5: print(f"top is: {peek(stack, top)}")
        case 6: print(f"size is: {sizeA(stack)}")
        case _ : print("invalid choice")
        
        
        